Grammar and Cleanup in Scala 3 Macro Documentation (#2121)
scala/docs.scala-lang: _overviews/scala3-macros/tutorial/quotes.md

```

In general, the quote delays the execution while the splice makes it happen before the surrounding code.
This generalisation allows us to also give meaning to a `${ ... }` that is not within a quote. This evaluates the code within the splice at compile-time and places the result in the generated code.

> Due to some technical considerations, only non-quoted splices are allowed directly within `inline` definitions that we call a [macro][macros].

Reviewer: ```suggestion
Due to some technical considerations, only top-level splices are allowed directly within `inline` definitions that we call a [macro][macros].
```

Consider the following code:
```scala
def evalAndUse[T](x: Expr[T])(using Quotes) = '{
  val x2: T = $x // error
  ... // use x2
}
```

Here, we will get an error telling us that we are missing a contextual `Type[T]`.
Therefore, we can easily fix it by writing:
```scala
def evalAndUse[T](x: Expr[T])(using Type[T])(using Quotes) = '{
  val x2: T = $x
  ... // use x2
}
```
This code will be equivalent to this more verbose version:
```scala
def evalAndUse[T](x: Expr[T])(using t: Type[T])(using Quotes) = '{
  val x2: t.Underlying = $x
  ... // use x2
}
```
Note that `Type` has a type member called `Underlying` that refers to the type held within the `Type`; in this case, `t.Underlying` is `T`.
Even if we use the `Type` implicitly, is generally better to keep it contextual as some changes inside the quote may require it.
The less verbose version is usually the best way to write the types as it is much simpler to read.
In some cases, we will not statically know the type within the `Type` and will need to use the `t.Underlying` to refer to it.

When do we need this extra `Type` parameter?

> * When a type is abstract and it is used at a level that is deeper than the current level.

Reviewer: ```suggestion
* When a type is abstract and it is used at a level that is higher than the current level.
```

Note that this time we have added the `T` explicitly in the pattern, even though it could be inferred.
By referring to the generic type `T` in the pattern, we are required to have a given `Type[T]` in scope.
This implies that `$x: T` will only match if `x` is of type `Expr[T]`.
In this particular case, this condition will always be true.

Now consider the following variant where `x` is an optional value with a (statically) unknown element type:

```scala
def exprOfOptionOf[T: Type](x: Expr[Option[Any]])(using Quotes): Option[Expr[T]] =
  x match
    case '{ Some($x: T) } => Some(x) // x: Expr[T]
    case _ => None
```

> This time, the pattern `Some($x: T)` will only match if the type of the Option is `Some[T]`.

Reviewer: ```suggestion
This time, the pattern `Some($x: T)` will only match if the type of the `Option` is `Some[T]`.
```
